### PR TITLE
[lldb][swift] Only run Swift API tests when Swift support is enabled

### PR DESCRIPTION
**Pull request by Teemperor**
Disabling Swift support in LLDB doesn't prevent the test suite from running
the Swift tests (which then end up failing instead of being marked as
unsupported). This adds swift to the SBDebugger configuration and then checks
that Swift is enabled when running Swift API tests.

### CODE REVIEW

**Comment by Teemperor** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]
That seems like a good idea.

File: lldb/test/API/CMakeLists.txt [hunk 1]
```diff
@@ -58,8 +58,13 @@ set(LLDB_TEST_COMMON_ARGS
   -S nm
   -u CXXFLAGS
   -u CFLAGS
+  )
+if (LLDB_ENABLE_SWIFT_SUPPORT)
+  set(LLDB_TEST_COMMON_ARGS
+  ${LLDB_TEST_COMMON_ARGS}
   ${SWIFT_TEST_ARGS}
```
Comment (kastiglione) [inline lldb/test/API/CMakeLists.txt]: instead of `set(VAR ${VAR} ...)` use `list(APPEND VAR ...)`?

```suggestion
  list(APPEND LLDB_TEST_COMMON_ARGS ${SWIFT_TEST_ARGS})
```

**Review by kastiglione — APPROVED** [review]

File: lldb/source/API/SBDebugger.cpp [hunk 2]
```diff
@@ -702,6 +702,14 @@ SBStructuredData SBDebugger::GetBuildConfiguration() {
       "A boolean value that indicates if lua support is enabled in LLDB");
   AddLLVMTargets(*config_up);
 
+  bool swift_enabled = false;
```
Comment (JDevlieghere) [inline lldb/source/API/SBDebugger.cpp]: What happens if we wrap the AddBoolConfigEntry in the `ifdef` and don't set the entry when swift is disabled? If we upstream SWIFT it would be similar to `LLDB_ENABLE_PYTHON ` and we could just remove the ifdef wrapped around it.

File: lldb/test/API/CMakeLists.txt [hunk 3]
```diff
@@ -58,8 +58,13 @@ set(LLDB_TEST_COMMON_ARGS
   -S nm
   -u CXXFLAGS
   -u CFLAGS
+  )
+if (LLDB_ENABLE_SWIFT_SUPPORT)
```
Comment (JDevlieghere) [inline lldb/test/API/CMakeLists.txt]: Why is this needed in the first place? If the variable doesn't exist it will just be empty. I'm actually in the process of removing this variable altogether.

File: lldb/source/API/SBDebugger.cpp (hunk 2 above)
Comment (Teemperor) [reply lldb/source/API/SBDebugger.cpp]: That seems to also do the trick (even though I feel like looking up a config value that doesn't exist should maybe be a hard error? Otherwise having typos in the setting string will just silently disable the thing).

File: lldb/test/API/CMakeLists.txt (hunk 3 above)
Comment (Teemperor) [reply lldb/test/API/CMakeLists.txt]: Makes sense. The variable is actually always set (I still need to put all the swift test setup code behind an `if (swift)`) but passing in the swift test parameters doesn't seem to break anything and it will resolve itself once I get around to update the test CMake stuff.

File: lldb/test/API/CMakeLists.txt (hunk 1 above)
Comment (Teemperor) [reply lldb/test/API/CMakeLists.txt]: That is nicer but I completely removed my change (see Jonas comment).

**Comment by Teemperor** [comment]
@swift-ci test

**Comment by Teemperor** [comment]
swift-ci test platform macos

**Comment by Teemperor** [comment]
@swift-ci test platform macos

**Comment by Teemperor** [comment]
@swift-ci test platform macos

**Comment by Teemperor** [comment]
@swift-ci please test

**Comment by Teemperor** [comment]
@swift-ci please test
